src/Component/Clock.tsx

```diff
@@ -0,0 +1,65 @@
+import { Component } from 'react';
+
+type State = {
+  currentTime: string,
+};
+
+type Props = {
+  clockName: string,
+};
+
+function getCurrentTime() {
+  return new Date().toUTCString().slice(-12, -4);
+}
+
+export class Clock extends Component<Props, State> {
+  state: Readonly<State> = {
+    currentTime: getCurrentTime(),
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({
+        currentTime: getCurrentTime(),
+      });
+
+      console.info(this.state.currentTime);
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props): void {
+    if (prevProps.clockName !== this.props.clockName) {
+      console.debug(`Renamed from ${prevProps.clockName} to ${this.props.clockName}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
+
+  render(): React.ReactNode {
+    const {
+      currentTime,
+    } = this.state;
+
+    const {
+      clockName,
+    } = this.props;
```

Comment:
Can be shorter, but it's up to you
```suggestion
    const { currentTime } = this.state;
    const { clockName } = this.props;
```